src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -342,6 +342,25 @@ private function parseDefinition(string $id, $service, string $file, array $defa
             throw new InvalidArgumentException(sprintf('A service definition must be an array or a string starting with "@" but "%s" found for service "%s" in "%s". Check your YAML syntax.', get_debug_type($service), $id, $file));
         }
 
+        if (['stack'] === array_keys($service) && \is_array($service['stack']) && $service['stack']) {
+            $stackId = null;
+
+            foreach (array_reverse($service['stack'], true) as $i => $service) {
+                if (null === $stackId) {
+                    $stackId = $id;
+                } elseif (!\is_array($service)) {
+                    throw new InvalidArgumentException(sprintf('A stacked service definition must be an array but "%s" found at position "%s" for service "%s" in "%s". Check your YAML syntax.', get_debug_type($service), $i, $id, $file));
+                } else {
+                    $stackId = '.stack.'.$id.'.'.$i;
+                    $service['decorates'] = $id;
```

Comment:
manually configuring `decorates` should be disallowed for stacked services, isnt it?

Reply:
Ppl will figure out it doesn't make sense to define this in the middle of a stack. I'd better not add code to deal with this.